contracts/stake/src/wasm.rs

```diff
@@ -100,38 +103,43 @@ impl StakeContract {
 
         // verify signature
         let message = Self::unstake_sign_message(counter, note);
-        let valid_signature = rusk_abi::verify_bls_sign(signature, pk, message);
+        let valid_signature =
+            rusk_abi::verify_bls(message, APK::from(&pk), signature);
 
         if !valid_signature {
             panic!("Invalid signature!");
         }
 
         // make call to transfer contract to withdraw a note from this contract
         // containing the value of the stake
-        let transfer = rusk_abi::transfer_contract();
-        let transaction =
-            Call::withdraw_from_transparent(value, note, withdraw_proof)
-                .to_transaction();
-
-        dusk_abi::transact_raw(self, &transfer, &transaction, 0)
+        let transfer = rusk_abi::transfer_module();
+        let transaction = RawTransaction::new(
+            "unstake",
+            Wfct2 {
+                value,
+                note,
+                proof: withdraw_proof,
+            },
+        );
```

Comment:
This will not work, since the transfer contract does not export an `unstake` function. The function you want to call here is `wfct`. I also don't particularly get why we're using `transact_raw` here, since it would be easier and more readable to just use `transact`.
Also I don't understand the introduction of `Wfct2` here. The `Wfct` struct is already present and is exactly the same.